unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts

```diff
@@ -77,27 +104,50 @@ export class DgraphDataSource extends RESTDataSource {
     offset?: number,
     match?: FilterMatch
   ) {
-    const orderString = `${orderDesc ? 'orderdesc' : 'orderasc'}: ${orderField}`;
+    const typeString = this.getEntityTypeString(type);
+    const categoryString = this.getEntityCategoryString(type);
+    const orderString = this.getEntitiesOrderString(orderField, orderDesc);
     const filterString = this.getEntitiesFilterString(description, match);
 
     return `{
-      all as counters(func: anyofterms(type, "${type}")) ${filterString} { 
+      all as counters(func: eq(dgraph.type, "${typeString}")) ${filterString} @cascade { 
+        ~children @filter(eq(name, "${categoryString}"))
         total: count(uid)
       }
       
       query(func: uid(all), ${orderString}, offset: ${offset}, first: ${first})  {
         code
-        description
-        type
+        description: name@*
+        type: dgraph.type
         uid
-        properties: has_property {
+        properties {
           type
           value
         }
       }
     }`;
   }
 
+  private static getEntityType(entity: IEntity): string {
+    const { type: types } = entity;
+    const [type] = types ?? [];
+  
+    switch (type) {
+      case "Product":
+        return "drug";
```

Comment:
Will probably eventually need to add `consumable`/`other` handling here but I think not a blocker for this PR - (since these weren't available in V1 of the API anyway 😸)